UnitTests/ReviewServiceTests.cs

```diff
@@ -0,0 +1,192 @@
+            };
+            //Act
+            await service.AddReview(review);
+            var reviews = await service.GetReviewsAwaitingApproval();
+            //Assert
+            Assert.IsNotNull(reviews);
+            Assert.That(reviews, Has.Count.EqualTo(2));
+        }
+
+        [Test, Order(3)]
+        public async Task ApproveReview()
+        {
+            //Arrange
+            using var context = new ApplicationDbContext(ContextOptions);
+            var service = new ReviewService(context);
+            var reviews = await service.GetReviewsAwaitingApproval();
+            var review = reviews[1];
+            //Act
+            await service.ApproveReview(review);
+            var approvedReview = context.Reviews.Where(a => a.Approved == true).ToList();
+            //Assert
+            Assert.IsNotNull(approvedReview);
+            Assert.That(approvedReview, Has.Count.EqualTo(1));
+        }
+
+        [Test, Order(4)]
+        public async Task RejectReview()
+        {
+            //Arrange
+            using var context = new ApplicationDbContext(ContextOptions);
+            var service = new ReviewService(context);
+            var reviews = await service.GetReviewsAwaitingApproval();
+            var review = reviews[0];
+            //Act
+            await service.RejectReview(review);
+            var approvedReview = context.Reviews.Where(a => a.Approved == false).ToList();
+            //Assert
+            Assert.IsNotNull(approvedReview);
+            Assert.That(approvedReview, Has.Count.EqualTo(1));
+        }
+
+        [Test, Order(5)]
+        public async Task GetApprovedReviews()
+        {
+            //Arrange
+            using var context = new ApplicationDbContext(ContextOptions);
+            var service = new ReviewService(context);
+            //Act
+            var approvedReviews = await service.GetApprovedGameReviews(testGame.GameId);
+            //Assert
+            Assert.IsNotNull(approvedReviews);
+            Assert.That(approvedReviews, Has.Count.EqualTo(1));
+        }
+
+        [Test, Order(6)]
+        public async Task GetReviewsByUser()
+        {
+            //Arrange
+            using var context = new ApplicationDbContext(ContextOptions);
+            var service = new ReviewService(context);
+            //Act
+            var userReviews = await service.GetReviewsByUser(users[0].Id);
+            //Assert
+            Assert.IsNotNull(userReviews);
+            Assert.That(userReviews, Has.Count.EqualTo(1));
+        }
+
+        [Test, Order(7)]
+        public async Task GetAverage()
+        {
+            //Arrange
+            using var context = new ApplicationDbContext(ContextOptions);
+            var service = new ReviewService(context);
+            //Act
+            var average = await service.GetAverageRating(testGame.GameId);
+            //Assert
+            Assert.IsNotNull(average);
+            Assert.AreEqual(3, average);
```

Comment:
If you are comparing 2 doubles, even if they are whole numbers, it is recommended that you follow the `abs(actual - expected) < threshold` style. [Here's](https://docs.nunit.org/articles/nunit/writing-tests/constraints/EqualConstraint.html#comparing-floating-point-values) an example

Reply:
I added the within method using the documentation you sent